spyderlib/widgets/tabs.py

```diff
@@ -112,7 +135,8 @@ def __init__(self, parent, actions=None, menu=None,
         QTabWidget.__init__(self, parent)
         
         self.setUsesScrollButtons(True)
-        
+        self.filter = TabFilter(parent)  # Need to keep reference

```

Comment:
The thing is that I had to do this incrementally... and as I escalated I had to rearrange and move stuff... it was very tricky!!